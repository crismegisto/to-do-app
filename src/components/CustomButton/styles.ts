import { StyleSheet } from 'react-native';

const DEFAULT_SPACE = 10;

const styles = StyleSheet.create({
  button: {
    width: DEFAULT_SPACE * DEFAULT_SPACE,
    alignItems: 'center',
    backgroundColor: 'blue',
    padding: DEFAULT_SPACE,
    borderRadius: DEFAULT_SPACE
  },
  text: {
    color: 'white',
    fontWeight: 'bold'
  }
});

export default styles;
