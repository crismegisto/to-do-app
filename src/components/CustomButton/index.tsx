import React from 'react';
import { Text, TouchableOpacity } from 'react-native';

import styles from './styles';

interface Props {
  name: string;
  onSubmit: () => void;
}

function CustomButton({ name, onSubmit }: Props) {
  return (
    <TouchableOpacity style={styles.button} onPress={onSubmit}>
      <Text style={styles.text}>{name}</Text>
    </TouchableOpacity>
  );
}

export default CustomButton;
