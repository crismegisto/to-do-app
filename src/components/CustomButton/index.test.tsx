import React from 'react';
import { screen } from '@testing-library/react-native';

import { renderWithProvider, initialStoreState } from '../../../utils/test/redux';
import CustomButton from './index';

describe('<CustomButton /> Tests', () => {
  it('match snapshot and button name', () => {
    const buttonName = 'Add Task';

    renderWithProvider(<CustomButton name={buttonName} onSubmit={() => jest.fn()} />, {
      preloadedState: initialStoreState
    });

    expect(screen.toJSON()).toMatchSnapshot();
    expect(screen.getByText('Add Task')).toBeOnTheScreen();
  });
});
