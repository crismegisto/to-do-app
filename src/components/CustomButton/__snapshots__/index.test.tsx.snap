// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> Tests match snapshot and button name 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "blue",
      "borderRadius": 10,
      "opacity": 1,
      "padding": 10,
      "width": 100,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontWeight": "bold",
      }
    }
  >
    Add Task
  </Text>
</View>
`;
